### DISCUSSION TITLE
rename files based on MIME type

### DISCUSSION THREAD

**Discussion by chbechmann** [ImageMagick/ImageMagick#2150] [Help]
TL: DR
I need to rename 10.000+ files based on their MIME-type, so they have the right extension.

Long read:
Hi, newbie here.

I am helping my friend moving data from one database to another. For that, he has a load of images (10.000+) which needs to be sorted by file type (jpg, png and TGA).

Problem is, that the files are only available without file-extensions. They are all named "productserial.angle", angle being a number between 1 and 3, eg. "05047294014.1". I would like this to become "05047294014.1.jpg" (if it is a jpg, of course)

The new database only accepts jpeg-encoded files, so I need some way to sort the three file types. Therefore, my best idea would be to use ImageMagick to read the MIME type and then append the correct extension or another unique identifier.

Is this possible? When using "magick identify *.*", I can get the relevant info, I just don't understand what syntax I need to use in order to append the information to the file name.

Example of identify command:
73926348.1 PNG 600x596 600x596+0+0 8-bit sRGB 248.08KiB 0.000u 0:00.000
78953239.2 PNG 600x599 600x599+0+0 8-bit sRGB 692.51KiB 0.000u 0:00.000

Thanks for any help!

**Comment by fmw42** [comment]
`magick identify -format "%m" image`

will provide the file type such as PNG or JPG from the header. See https://imagemagick.org/script/escape.php

**Comment by chbechmann** [comment]
Hey.
I tried your string, but the output is a bit cryptic. It simply lists all the extensions after each other. In a folder with 3 files it returns "JPEGPNGPNG", but no files were renamed.

For other readers:
I did find a workaround, but using ExifTool instead.
https://exiftool.org/forum/index.php?topic=11327.msg60759#msg60759

**Reply by fmw42** [reply]
Put \n in the command as:

`magick identify -format "%m\n" image
`
That will separate the result for each image on a separate line

**Comment by snibgo** [comment]
IM can't rename files.

IM can tell you, with "%m", what the extension should be. You could use a longer format that wrote, for each file, the shell command that renames the file.
